pkg/azure/aks/price_store.go

```diff
@@ -80,7 +54,21 @@ func NewPricingStore(subId string, priceClient *retailPriceSdk.RetailPricesClien
 		}
 	}()
 
-	return p
+	return p, err
+}
+
+func (p *PriceStore) getPriceInfoFromVmInfo(vmInfo *VirtualMachineInfo) (float64, error) {
+	p.regionMapLock.RLock()
+	defer p.regionMapLock.RUnlock()
+
+	region := vmInfo.Region
+	priority := vmInfo.Priority
+	operatingSystem := vmInfo.OperatingSystem
+	sku := vmInfo.MachineTypeSku
+
+	vmPriceInfo := p.RegionMap[region][priority][operatingSystem][sku]
+
+	return vmPriceInfo.RetailPrice, nil
```

Comment:
I don't ever see a scenario where an error is returned. However there's quite a few places where a panic could be introduced. Suggestion: Check for the existence of keys, and return error if say:
1. region isn't found
2. priority class isn't found
3. operating system isn't found
4. sku isn't found